src/components/coin.ts

```diff
@@ -269,3 +274,191 @@ export const applyBonusByUserId = async (userId: string): Promise<boolean> => {
   }
   return false;
 };
+
+export enum TransferSign {
+  Pending = 0,
+  Accept = 1,
+  Decline = 2,
+}
+
+export const getEmojiFromSign = (sign: TransferSign): string => {
+  switch (sign) {
+    case TransferSign.Pending:
+      return '❓';
+    case TransferSign.Accept:
+      return '✅';
+    case TransferSign.Decline:
+      return '❌';
+  }
+};
+
+export enum TransferResult {
+  Pending,
+  Rejected,
+  Confirmed,
+}
+
+type TransferState = {
+  sender: User;
+  receiver: User;
+  result: TransferResult;
+  amount: number;
+  reason?: string;
+};
+
+class TransferTracker {
+  transfers: Map<string, Transfer>; // id, transfer
+
+  constructor() {
+    this.transfers = new Map<string, Transfer>();
+  }
+  getTransferFromId(id: string): Transfer | undefined {
+    return this.transfers.get(id);
+  }
+
+  runFuncOnGame(transferId: string, func: (transfer: Transfer) => void): void {
```

Comment:
Change this method call into something else, since I think a `Game` does not make sense in this context. Maybe `runFuncOnTransfer`?